lib/p2p/Pool.ts

```diff
@@ -73,6 +74,50 @@ class Pool extends EventEmitter {
       return;
     }
 
+    if (this.listenPort) {
+      if (!handshakeData.addresses) {
+        handshakeData.addresses = [];
+      }
+
+      // Append the external IP if no address was specified by the user
+      if (handshakeData.addresses.length === 0) {
+        try {
+          // TODO: verify that this address is reachable
+          const externalip = await new Promise<string>((resolve, reject) => {
+            http.get('http://ipv4.icanhazip.com/', (res) => {
+              let body = '';
+
+              res.on('data', (chunk) => {
+                body += chunk;
+              });
+              res.on('end', () => {
+                // Removes new line at the end of the string
+                body = body.trimRight();
+                resolve(body);
+              });
+              res.on('error', (err: Error) => {
+                reject(errors.EXTERNAL_IP_UNRETRIEVABLE(err));
+              });
+
+            }).on('error', (err: Error) => {
+              reject(errors.EXTERNAL_IP_UNRETRIEVABLE(err));
+            });
+          });
+
+          this.logger.verbose(`retrieved external IP: ${externalip}`);
+
+          handshakeData.addresses.push({
+            host: externalip,
+            port: this.listenPort,
+          });
+
+        } catch (error) {
+          this.logger.error(error.message);
+          return;
```

Comment:
I don't think we'd want to return here, because that would abort the init process halfway. I think we should probably just catch, log, and continue without any external addresses. The alternative would be not to catch here and just let the error get thrown, potentially aborting the server (although I think currently we catch everything in the Xud.start() method, but probably that should change at some point).